ralph-wiggum-toolkit/core/scripts/lib.sh

```
@@ -0,0 +1,452 @@
+
+  local parent_desc parent_spec
+  IFS=$'\t' read -r parent_desc parent_spec <<< "$parent_data"
+
+  write_state --arg fid "$fix_id" --arg pid "$parent_id" \
+    --arg desc "Fix Tier $tier failures for $parent_id" \
+    --arg spec "$parent_spec" \
+    --arg acc "All Tier $tier quality gates pass" \
+    '.tasks += [{
+      id: $fid, description: $desc, spec: $spec, acceptance: $acc,
+      status: "pending", dependencies: [], attempts: 0, parentId: $pid
+    }] | .currentTaskId = $fid | .taskIteration = 1'
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)